tt_eager/tt_dnn/op_library/sdpa/kernels/compute/sdpa_flash_decode.cpp

```diff
@@ -0,0 +1,706 @@
+                max_block(cb_m_in, cb_prev_max, cb_cur_max, Sq_chunk_t); // pushed, pushed, popped
+
+                // copy_block(cb_prev_max, cb_cur_max, Sq_chunk_t);
+                // DPRINT << "[C] R ckpt 1.1" << ENDL();
+                // cb_push_back(cb_prev_max, Sq_chunk_t);
+                // DEBUG_STATUS("XCAA");
+                // DPRINT << "[C] R ckpt 1.2" << ENDL();
+                // max_block_inplace<cb_cur_max, cb_prev_max_2, Sq_chunk_t>(); // TODO: NEED TO FIX THIS
+
+                // DPRINT_MATH(DPRINT << "[C] R ckpt 2 math" << ENDL());
+                // DPRINT_PACK(DPRINT << "[C] R ckpt 2 pack" << ENDL());
+                // DPRINT_UNPACK(DPRINT << "[C] R ckpt 2 unpack" << ENDL());
+                // DEBUG_STATUS("XCAB");
+
+                // l = torch.exp(m_2 - m) * l_2 + torch.exp(m_1 - m) * l_1
+                /// l1 = torch.exp(m_2 - m) * l_2
+                // unpack_reconfig_data_format(cb_prev_max_2, cb_cur_max); // DEBUG
+                // pack_reconfig_data_format(cb_exp_max_diff_2);
+                sub_exp_block(cb_m_in, cb_cur_max, cb_exp_max_diff_2, Sq_chunk_t);
+                mul_block_inplace(cb_prev_sum_2, cb_exp_max_diff_2, Sq_chunk_t);
+                /// l2 = torch.exp(m_1 - m) * l_1
+                // unpack_reconfig_data_format(cb_prev_max, cb_cur_max); // DEBUG
+                // pack_reconfig_data_format(cb_exp_max_diff);
+                sub_exp_block(cb_prev_max, cb_cur_max, cb_exp_max_diff, Sq_chunk_t);
+                mul_block_inplace(cb_prev_sum, cb_exp_max_diff, Sq_chunk_t);
+                /// l = l1 + l2
+                // unpack_reconfig_data_format(cb_cur_sum, cb_prev_sum); // DEBUG
+                // pack_reconfig_data_format(cb_cur_sum);
+                add_block(cb_prev_sum_2, cb_prev_sum, cb_cur_sum, Sq_chunk_t);
+
+                // DPRINT << "[C] R ckpt 3" << ENDL();
+
+                // O = torch.matmul(torch.eye(padded_num_heads) * torch.exp(m_2 - m), O_2) + torch.matmul(torch.eye(padded_num_heads) * torch.exp(m_1 - m), O_1)
+                /// O_1 = torch.matmul(torch.eye(padded_num_heads) * torch.exp(m_2 - m), O_2)
+                // unpack_reconfig_data_format(cb_out_accumulate_im_2, cb_exp_max_diff_2); // DEBUG
+                // pack_reconfig_data_format(cb_out_accumulate_im_2);
+                // UNPACK(TTI_STALLWAIT(p_stall::STALL_UNPACK, p_stall::UNPACK));
+                // UNPACK(TTI_STALLWAIT(p_stall::STALL_CFG, p_stall::UNPACK));
+                // unpack_reconfig_data_format_srca(cb_out_accumulate_im_2); // DEBUG
+                // UNPACK(TTI_STALLWAIT(p_stall::STALL_UNPACK, p_stall::TRISC_CFG));
+                // // MATH(llk_math_reconfig_data_format_srca(cb_out_accumulate_im_2));
+                // // unpack_reconfig_data_format_srcb(cb_exp_max_diff_2); // DEBUG
+                // // UNPACK(tensix_sync());
+                // UNPACK(TTI_STALLWAIT(p_stall::STALL_UNPACK, p_stall::UNPACK));
+
```

Comment:
I have left the comments here because there is still an existing nd pcc bug that's being worked around but not solved. I expect to work on it again soon after there are progress on those issues. What do you think? Should I remove those comments?

Reply:
Ok that makes sense. Let's leave these comments until ND is resolved